test/test-rpm-macros.sh

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# This test makes some basic checks that RPM macros work correctly.
+# RPM is a simple C program available on different Linux distros, not only RPM-based ones,
+# and even BSD systems, so it must not be a problem to require it.
+# rpmspec utility is required (so this test will work with RPM 4 but won't work with RPM 5).
+set -eu
+
+BUILD_DIR="${1:?Missing argument: build directory}"
+RPM_MACROS_FILE="${BUILD_DIR:?}/src/rpm/macros.systemd"
+
+if ! command -v rpm >/dev/null || ! command -v rpmspec >/dev/null; then
+        echo >&2 "Missing necessary utilities (rpm, rpmspec), can't continue"
+        exit 1
+fi
+
+if [[ ! -f "${RPM_MACROS_FILE:?}" ]]; then
+        echo "RPM macros file not found in $RPM_MACROS_FILE!"
+        exit 1
+fi
+
+at_exit() {
+    if [[ -v WORK_DIR && -d "$WORK_DIR" ]]; then
+        rm -frv "$WORK_DIR"
+    fi
+}
+
+trap at_exit EXIT
+
+WORK_DIR="$(mktemp -d)"
+RPM_SPEC="$(mktemp "$WORK_DIR/systemd-test-rpm-macros-XXX.spec")"
+TEMP_LOG="$(mktemp "$WORK_DIR/out-XXX.log")"
+
+die() {
+    echo >&2 "${1:?}"
+    exit 1
+}
+
+mk_mini_spec() {
```

Comment:
I feel like this function should take a param, instead of using `echo "" >> $RPM_SPEC` everywhere.